mainwindow.cc

```diff
@@ -2207,85 +2204,74 @@ void MainWindow::editPreferences()
-
-    bool needReload = false;
+    newCfg.zoomFactor = cfg.preferences.zoomFactor;
+    newCfg.helpZoomFactor = cfg.preferences.helpZoomFactor;
+    newCfg.wordsZoomLevel = cfg.preferences.wordsZoomLevel;
+    newCfg.hideMenubar = cfg.preferences.hideMenubar;
+    newCfg.searchInDock = cfg.preferences.searchInDock;
+    newCfg.alwaysOnTop = cfg.preferences.alwaysOnTop;
+
+    newCfg.proxyServer.systemProxyUser = cfg.preferences.proxyServer.systemProxyUser;
+    newCfg.proxyServer.systemProxyPassword = cfg.preferences.proxyServer.systemProxyPassword;
+
+    newCfg.fts.dialogGeometry = cfg.preferences.fts.dialogGeometry;
+    newCfg.fts.matchCase = cfg.preferences.fts.matchCase;
+    newCfg.fts.maxArticlesPerDictionary = cfg.preferences.fts.maxArticlesPerDictionary;
+    newCfg.fts.maxDistanceBetweenWords = cfg.preferences.fts.maxDistanceBetweenWords;
+    newCfg.fts.searchMode = cfg.preferences.fts.searchMode;
+    newCfg.fts.useMaxArticlesPerDictionary = cfg.preferences.fts.useMaxArticlesPerDictionary;
+    newCfg.fts.useMaxDistanceBetweenWords = cfg.preferences.fts.useMaxDistanceBetweenWords;
+    newCfg.fts.ignoreWordsOrder = cfg.preferences.fts.ignoreWordsOrder;
+    newCfg.fts.ignoreDiacritics = cfg.preferences.fts.ignoreDiacritics;
 
     // See if we need to reapply Qt stylesheets
-    if( cfg.preferences.displayStyle != p.displayStyle ||
-      cfg.preferences.darkMode != p.darkMode )
+    if( cfg.preferences.displayStyle != newCfg.displayStyle ||
+      cfg.preferences.darkMode != newCfg.darkMode )
     {
-      applyQtStyleSheet( p.addonStyle, p.displayStyle, p.darkMode );
+      applyQtStyleSheet( newCfg.addonStyle, newCfg.displayStyle, newCfg.darkMode );
     }
 
-    // see if we need to reapply articleview style
-    if( cfg.preferences.displayStyle != p.displayStyle ||
-      cfg.preferences.addonStyle != p.addonStyle ||
-      cfg.preferences.darkReaderMode != p.darkReaderMode )
-    {
-      articleMaker.setDisplayStyle( p.displayStyle, p.addonStyle );
-      needReload = true;
-    }
-
-    if( cfg.preferences.collapseBigArticles != p.collapseBigArticles
-        || cfg.preferences.articleSizeLimit != p.articleSizeLimit )
-    {
-      articleMaker.setCollapseParameters( p.collapseBigArticles, p.articleSizeLimit );
-    }
+    bool needReload = ( cfg.preferences.displayStyle != newCfg.displayStyle ||
+                        cfg.preferences.addonStyle != newCfg.addonStyle ||
+                        cfg.preferences.darkReaderMode != newCfg.darkReaderMode );
 
     // See if we need to reapply expand optional parts mode
-    if( cfg.preferences.alwaysExpandOptionalParts != p.alwaysExpandOptionalParts )
+    if( cfg.preferences.alwaysExpandOptionalParts != newCfg.alwaysExpandOptionalParts )
     {
-      emit setExpandOptionalParts( p.alwaysExpandOptionalParts );
+      emit setExpandOptionalParts( newCfg.alwaysExpandOptionalParts );
       // Signal setExpandOptionalParts reload all articles
       needReload = false;
     }
 
```

Comment:
this seems a hidden bug in the previous code.

if `displayStyle` and `alwaysExpandOptionalParts` both changed ,`needReload` will be set to `false`

Reply:
Yes, there is a bug here.